### PR TITLE
People year

### LINKED ISSUE #617: Link years on people pages to the year pages
Click this, go to year

image

### LINKED ISSUE #1225: Graduated person appearing as student
End user submitted issue from page: [/people/ramsay_gilderdale/](https://history.newtheatre.org.uk/people/ramsay_gilderdale/)
---
He appears as a student when he quite clearly isn't
*NP*

### PR DESCRIPTION
**Pull request by samozzy**
Add links to a person's graduation year on the person page. 
To close #617 and to close #1225

### CODE REVIEW

File: _layouts/person.html [hunk 1]
```diff
@@ -4,6 +4,9 @@
 ---
 {% assign person = page %}
 {% capture submit_link %}/collect/person/?name={{person.path_name}}{% endcapture %}
+{% if person.graduated %}
+  {% capture year_link %}/years/{{ person.graduated | minus: 1 | slice: 2,2 }}_{{ person.graduated | slice: 2,2 }}/{% endcapture %}
```
Comment (wjdp) [inline _layouts/person.html]: Can we put this in `person.rb`? Hesitant to use liquid for stuff like this as it tends to break in weird ways when we upgrade.

File: _layouts/person.html [hunk 2]
```diff
@@ -29,6 +32,7 @@ <h1 class="person-name">{{ person.title | escape_once }}</h1>
     </div>
 
     <div class="person-graduated">
+      {% if person.graduated %}{% unless person.student == true %}<a href="{{ year_link }}">{% endunless %}{% endif %}
```
Comment (wjdp) [inline _layouts/person.html]: `{% if person.graduated and person.student == false %}`

File: _layouts/person.html (hunk 1 above)
Comment (samozzy) [reply _layouts/person.html]: Done.

File: _layouts/person.html (hunk 2 above)
Comment (samozzy) [reply _layouts/person.html]: No longer needed as using `{% if person.grad_link %}`

**Comment by samozzy** [comment]
To do: make the graduation year not appear in search results for those who have not graduated yet. (Thanks @BenWoodford76)

**Comment by samozzy** [comment]
@wjdp I think everything is now working and up to scratch here. Could you verify please?

**Pull request merged**